en/news/_posts/2015-11-27-ruby-2-3-0-preview2-released.md

```diff
@@ -0,0 +1,81 @@
+---
+layout: news_post
+title: "Ruby 2.3.0-preview2 Released"
+author: "naruse"
+translator:
+date: 2015-11-11 14:00:00 +0000
+lang: en
+---
+
+We are pleased to announce the release of Ruby 2.3.0-preview2.
+
+Ruby 2.3.0-preview2 is the second preview of Ruby 2.3.0.
+Many new features and improvements are included.
+
+A [Frozen String Literal Pragma](https://bugs.ruby-lang.org/issues/11473)
+is introduced.
+With Ruby 2.1, `"str".freeze` has been optimized to reduce object allocation.
+Ruby 2.3 introduces a new magic comment and command line option to
+freeze all string literals in the source files.
+Additionally for debugging, you can get where the object is created on
+`"can't modify frozen String"` error by
+`--enable-frozen-string-literal-debug`.
+
+A [safe navigation operator](https://bugs.ruby-lang.org/issues/11537)
+[`&.`](https://instagram.com/p/-M9l6mRPLR/),
+which already exists in C#, Groovy, and Swift, is introduced to ease
+`nil` handling as `obj&.foo`. `Array#dig` and `Hash#dig` are also added.
+
+The [did_you_mean gem is bundled](https://bugs.ruby-lang.org/issues/11252).
+The did_you_mean gem
+shows the candidates on the `NameError` and `NoMethodError` to ease
+debugging.
+
+Try and enjoy programming with Ruby 2.3.0-preview2, and report us your
+knowledge!
+
+## Notable Changes since 2.2
+
+* TBD
```

Comment:
Ruby 2.3 includes many performance tuning fixes.
For example, [optimize Proc#call] (https://bugs.ruby-lang.org/issues/11569),
[reconsider method entry data structure] (https://bugs.ruby-lang.org/issues/11278),
[introducing new table data structure] (https://bugs.ruby-lang.org/issues/11420),
machine code level tuning for object allocation and method calling code,
and other many optimizations.